### PR TITLE
feat: list all pages

### PR DESCRIPTION
**Pull request by kindermoumoute**

### CODE REVIEW

File: scw/lister.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package scw
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+type lister interface {
```
Comment (jerome-quere) [inline scw/lister.go]: this is not the interface we talk about.

File: scw/lister.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+// doListAll collect all pages of a List request and aggregate all results on a single response.
+func (c *Client) doListAll(req *ScalewayRequest, res interface{}) (err error) {
+	switch response := res.(type) {
+	case lister:
+		n := 0
+		page := 1 // start at page 1
+		totalCount := -1
+		for i := 0; i < totalCount; i += n {
```
Comment (jerome-quere) [inline scw/lister.go]: will never enter the loop

File: scw/lister.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			// set current page
+			req.Query.Set("page", strconv.Itoa(page))
```
Comment (jerome-quere) [inline scw/lister.go]: We should avoid mutate the request (make a copy ?)

File: api/lb/v1/lb_sdk.go [hunk 4]
```diff
@@ -707,6 +707,20 @@ type ListLbsResponse struct {
 	TotalCount uint32 `json:"total_count,omitempty"`
 }
 
+func (r *ListLbsResponse) GetTotalCount() int {
```
Comment (DenBeke) [inline api/lb/v1/lb_sdk.go]: Is this generated code?

Comment (kindermoumoute) [reply api/lb/v1/lb_sdk.go]: Soon-to-be

Comment (DenBeke) [reply api/lb/v1/lb_sdk.go]: > Soon-to-be

👌

Just don't forget the Godoc 😜

File: scw/lister.go (hunk 3 above)
Comment (kindermoumoute) [reply scw/lister.go]: I can copy the query parameters at the beginning and set it back at the end, does it work for you?

File: scw/client.go [hunk 5]
```diff
@@ -110,16 +117,35 @@ func (c *Client) GetDefaultPageSize() (int32, bool) {
 	return 0, false
 }
 
-// Do performs an HTTP request based on the ScalewayRequest object.
-// RequestOptions are executed on the ScalewayRequest.
+// Do perform HTTP request(s) based on the ScalewayRequest object.
```
Comment (DenBeke) [inline scw/client.go]: perform**s**

**Comment by DenBeke** [comment]
LGTM

**Pull request merged**